client/src/parserFunctions.js

```diff
@@ -0,0 +1,38 @@
+import axios from 'axios';
+
+const parserFunctions = {
+  getRelatedItems: (productId, cachedProducts) => {
+    return axios.get(`/products/${productId}/related`)
+      .then(({data}) => {
+        let relatedProducts = data.map(product => {
+          if (cachedProducts[product]) {
+            return ['find'];
+          } else {
+            return axios.get(`./products/${product}`)
+              .then(({data}) => {
+                let productInfo = {
+                  currentProduct: data[0],
+                  productStylesArray: data[1].results,
+                  reviews: {
+                    reviewsMetadata: data[4],
+                    reviews: data[3],
+                  },
+                  questionList: data[2]
+                };
+                return productInfo;});
+          }
+        });
+        return Promise.all(relatedProducts);
+      })
+      .catch(err => {
+        alert(`Error encountered: ${err}`);
```

Comment:
It might be helpful to add a little more description, like Error encountered in getRelatedItems, but I guess the stack trace could tell you that too

Reply:
The alert is for the user. I don't know if it's relevant for them to understand where the error occurred.
(Usually, websites render a new page showing the error code, but they're not very descriptive)